src/base/settingsstorage.h

```diff
@@ -62,4 +62,27 @@ public slots:
     mutable QReadWriteLock m_lock;
 };
 
+template<typename T>
+class StorableValue
```

Comment:
>But CachedSettingValue is enough. 

No, it's not, at least not in current form, see `m_storeDialogSize` in this PR.
I'll need to add:
```c++
T CachedSettingValue::operator()(const T &defaultValue) const
{
    return (m_value != T()) ? m_value : defaultValue;
}
```